controller/common/manage_children.go

```diff
@@ -52,6 +52,9 @@ func ApplyUpdate(orig, update *unstructured.Unstructured) (*unstructured.Unstruc
 	if err := revertObjectMetaSystemFields(newObj, orig); err != nil {
 		return nil, fmt.Errorf("failed to revert ObjectMeta system fields: %v", err)
 	}
+	if err := revertStatus(newObj, orig); err != nil {
```

Comment:
Add a comment above this explaining that we revert status because we don't currently support a parent changing the status of its children, so we need to make sure no diffs on children involve status.